lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -449,19 +458,28 @@ private JourneyResponse buildReuseResponse(
                     : JOURNEY_OPERATIONAL_PROFILE_REUSE;
         }
 
+        var credentials = VcHelper.filterVCBasedOnProfileType(vcs, attainedVot.getProfileType());
         sessionCredentialsService.persistCredentials(
-                VcHelper.filterVCBasedOnProfileType(vcs, attainedVot.getProfileType()),
-                auditEventUser.getSessionId(),
-                false);
+                credentials, auditEventUser.getSessionId(), false);
+        migrateCredentialsToEVCS(auditEventUser.getUserId(), credentials);
 
         return JOURNEY_REUSE;
     }
 
+    private void migrateCredentialsToEVCS(String userId, List<VerifiableCredential> credentials)
```

Comment:
You might need to coordinate with @sam803 on this, but we need an additional check here, to make sure the user is not _already migrated_.

That is, if there are any `CURRENT` VCs in EVCS then we don't need to migrate.

Reply:
Oh I see - that's what the `filter` is for 🤔

I think we probably still need to think carefully about how this is going to work when we add EVCS reads.